toolchain/check/import_ref.cpp

```diff
@@ -83,12 +100,56 @@ class ImportRefResolver {
     return const_id;
   }
 
-  // Returns the ConstantId for a TypeId, or adds it to the stack and
-  // returns Invalid if the ConstantId is not ready.
+  // Returns the ConstantId for a TypeId. Adds it to the stack and sets
+  // has_unresolved to true if it's not ready.
   auto GetConstantId(SemIR::TypeId type_id) -> SemIR::ConstantId {
     return GetConstantId(import_ir_.types().GetInstId(type_id));
   }
 
+  // Given a param_refs_id, returns the necessary constants to convert it. Sets
+  // has_unresolved to true if any aren't ready.
```

Comment:
```suggestion
  // Returns the ConstantId for the type of each parameter in the parameter list param_refs_id, in order. Sets
  // has_unresolved to true if any aren't ready.
```

That's just a guess; it may be inaccurate. The main point of my suggestion is to clarify what the return value means (rather than what it's used for), and in particular to make it explicit that the elements of the return value correspond to the parameters.

Reply:
Rephrased, but focusing more on the API (that it's used by the other function) than the implementation (that it's iterating through the refs and only grabs the type, which seems more like an implementation detail).

Reply:
Let me put it a different way: the problem I'm trying to address is that I couldn't make sense of the body of `GetParamRefsId` except by also reading the body of `GetParamConstantIds`. That sort of coupling between the internals of separate functions makes the code substantially harder for me to read, so one of the main things I look for in API comments is that they enable me to reason locally about a single function body at a time.

I see what you're getting at, in the sense that this information is a sort of joint implementation detail of `GetParamConstantIds` and `GetParamRefsId` that can be hidden from the mutual caller that acts as an intermediary, but the contract between them still needs to be documented somewhere. There are ways of doing that while still hiding it from other functions (such as factoring out a separate class that it can be an internal invariant of), but I'm not sure that degree of complexity is necessary here.

Reply:
I've tried rephrasing this again, can you take another look?